File: lld_game_design_primer/blackjack_deck_of_cards/Approach_1/BlackJackCard.java
package lld_game_design_primer.blackjack_deck_of_cards.Approach_1;

public class BlackJackCard extends Card {
    public BlackJackCard(int c, SUIT s) {
        super(c, s);
    }

    public int value() {
        if (isAce()) { // Ace
            return 1;
        } else if (isFaceCard()) { // Face card
            return 10;
        } else { // Number card
            return faceValue;
        }
    }

    public int minValue() {
        return value();
    }

    public int maxValue() {
        if (isAce()) { // Ace
            return 11;
        } else {
            return value();
        }
    }

    public boolean isAce() {
        return faceValue == 1;
    }

    public boolean isFaceCard() {
        return faceValue >= 11 && faceValue <= 13;
    }
}

File: lld_game_design_primer/blackjack_deck_of_cards/Approach_1/BlackJackCard.py
from BlackJackCard import Card


class BlackJackCard(Card):
    def __init__(self, c, s):
        super().__init__(c, s)

    def value(self):
        if self.isAce():
            return 1
        elif self.isFaceCard():
            return 10
        else:
            return self.faceValue

    def minValue(self):
        return self.value()

    def maxValue(self):
        if self.isAce():
            return 11
        else:
            return self.value()

    def isAce(self):
        return self.faceValue == 1

    def isFaceCard(self):
        return self.faceValue >= 11 and self.faceValue <= 13